### PR TITLE
metrics server addon

### PR DESCRIPTION
**Pull request by pidah**
**What this PR does / why we need it**:
This PR includes an addon for the new kubernetes metrics server and enables it by default for kubernetes clusters version 1.9.0+. This is a requirement for Horizontal Pod Autoscaling as discussed here https://github.com/Azure/acs-engine/issues/2114

### CODE REVIEW

**Comment by jackfrancis** [comment]
Thanks @pidah!

**Comment by jackfrancis** [comment]
FYI source of rebase is this merge: https://github.com/Azure/acs-engine/commit/26ad11a4540cd5ac8938ccbd13064ddb81598821

(should be easy to resolve)

**Comment by pidah** [comment]
@jackfrancis thanks, rebased now

**Comment by jackfrancis** [comment]
@pidah will this change alone enable our existing E2E hpa tests to work w/ 1.9 clusters?

See:

https://github.com/Azure/acs-engine/blob/master/test/e2e/kubernetes/deployment/deployment.go#L173

tldr; we do a simple `kubectl autoscale deployment <name-of-deployment> --cpu-percent=5 --min=1 --max=10` and then hit the child pods w/ load and confirm that we have pod count growth

**Comment by jackfrancis** [comment]
Thanks for confirming @pidah, pushed up a change to enable this test coverage for 1.9 clusters 🤞

**Comment by jackfrancis** [comment]
@pidah Looks like it wasn't that simple. You should be able to review the results here:

https://circleci.com/gh/Azure/acs-engine/17395?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I'll have time next week to do some manual prodding with this cluster (build a 1.9 cluster from this PR, experiment with HPA). (Feel free to do your own manual prodding if you'd like to see this merge sooner.)

And thanks again so much!!

**Comment by pidah** [comment]
Thanks @jackfrancis
Will take a look tomorrow.

**Comment by pidah** [comment]
For some reason the the metrics-server APIService definition section https://github.com/DigitalInnovation/acs-engine/blob/1622b3c3ed0f1664fa4c8d6d3d671c3b30dd383f/parts/k8s/addons/kubernetesmasteraddons-metrics-server-deployment.yaml#L131-L146 is silently ignored -
```
apiVersion: apiregistration.k8s.io/v1beta1
kind: APIService
metadata:
  name: v1beta1.metrics.k8s.io
spec:
  service:
    name: metrics-server
    namespace: kube-system
  labels:
    kubernetes.io/cluster-service: "true"
    addonmanager.kubernetes.io/mode: EnsureExists
  group: metrics.k8s.io
  version: v1beta1
  insecureSkipTLSVerify: true
  groupPriorityMinimum: 100
  versionPriority: 100
```
Applying manually on the cluster post-creation works and HPA works fine. Played with `EnsureExists` & `Reconcile` labels but still no joy . Seen this before ? @jackfrancis

**Comment by pidah** [comment]
@jackfrancis fixed the labels...ready for testing now

**Comment by jackfrancis** [comment]
@pidah 🍻

**Comment by jackfrancis** [comment]
Test passed, awesome @pidah
